scripts/rate.sh

```diff
@@ -0,0 +1,31 @@
+#!/usr/bin/env bash
+
+### rate -- list all rates or compute specific ones
+### Usage:
+### ./rate.sh list all rates
+### ./rate.sh <entry> return the computed rate
+
+set -e
+
+rate() {
+    basispoints=$(printf "%.0f" "$(echo "scale=0; $1*100" | bc)")
+    normalizedamount="$( echo "scale=4; $basispoints/10000 + 1" | bc)"
+    rayte=$(bc -l <<< "scale=27; e( l($normalizedamount)/(60 * 60 * 24 * 365) ) * 10^27")
+    echo "$1%: ${rayte%.*}"
+}
+
+if [[ -z $1 ]];
+then
+    for n in $(seq 0 0.01 100);
+    do
+        rate "$n"
+    done
+fi
+
+if [[ $1 =~ ^([0-9]+)?([.][0-9]+)?$ ]];
```

Comment:
Your current regex is matching an empty string.  So if you run this with no arguments, it prints the list, then tosses an error after: 
```
Parse error: bad token
    <stdin>:1
```

This suggestion should do what you want.
```suggestion
if [[ $1 =~ ^([.]|[0-9]+)([.][0-9]+)?$ ]];
```

Reply:
oh! great catch! I took it from `mcd-duty` and testing I didn't noticed that edge case.

Reply:
I've improved the regex in 3450967 where it should match a non zero digit followed by zero or more digits with an optional decimal part with one or more digits.